src/transport_ofi.c

```diff
@@ -578,48 +598,65 @@ static inline int atomic_limitations_check(void)
-                    SHM_NAMED_BOPS[j], SHM_NAMED_DT_INT[i]);
-            if(!atomicwarn) {
-                OFI_ERRMSG("Error: atomicvalid ret=%d atomic_size=%d \n",
-                            ret, (int)atomic_size);
-	            return ret;
-            }
-        }
+         if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_BOPS[j], SHM_NAMED_DT_INT[i]))
+           return ret;
       }
     }
 
     /* OTHER OPS check */
-    for(i=0; i<5; i++) {//DT
+    for(i=0; i<6; i++) {//DT
       for(j=0; j<4; j++) { //OPS
         ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_CMP[i], SHM_OPS[j],
                         &atomic_size);
-        if(ret!=0 || atomic_size == 0) {
-            fprintf(stderr, "%s OFI detected no support for atomic '%s'"
-                    "on type '%s'\n", (atomicwarn ? "Warning" : "Error"),
-                    SHM_NAMED_OPS[j], SHM_NAMED_DT_CMP[i]);
-            if(!atomicwarn) {
-                OFI_ERRMSG("Error: atomicvalid ret=%d atomic_size=%d \n",
-                            ret, (int)atomic_size);
-	            return ret;
-            }
-        }
+         if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_OPS[j], SHM_NAMED_DT_CMP[i]))
+           return ret;
+      }
+    }
+
+    for(i=0; i<3; i++) {//DT does not include short
+      ret = fi_compare_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_INT[i], FI_CSWAP,
+                        &atomic_size);
+      if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                        "FI_CSWAP", SHM_NAMED_DT_INT[i]))
+        return ret;
+
+      ret = fi_fetch_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_INT[i], FI_SUM,
+                        &atomic_size);
+      if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                        "FI_SUM", SHM_NAMED_DT_INT[i]))
+        return ret;
+    }
+
+    ret = fi_compare_atomicvalid(shmem_transport_ofi_epfd, SHM_INTERNAL_INT, FI_MSWAP,
+                        &atomic_size);
+     if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                        "FI_MSWAP", "Int"))
+       return ret;
+
+    for(i=0; i<5; i++) {//DT
+      for(j=4; j<6; j++) { //OPS
+        ret = fi_fetch_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_CMP[i], SHM_OPS[j],
+                        &atomic_size);
+        if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_OPS[j], SHM_NAMED_DT_CMP[i]))
+          return ret;
       }
     }
 
     /* LONG DOUBLE limitation is common */
     for(j=0; j<4; j++) { //OPS
-        ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_INTERNAL_LONG_DOUBLE, SHM_OPS[j], &atomic_size);
-        if(ret!=0 || atomic_size == 0) {
-		shmem_transport_have_long_double = 0;
+      ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_INTERNAL_LONG_DOUBLE, SHM_OPS[j],
+                &atomic_size);
+      if(ret!=0 || atomic_size == 0) {
```

Comment:
so count = provider atomic size, but since I've got that value at this point I can do the assert, and agreed that should be added